dotCMS/src/main/java/com/dotcms/contenttype/business/UniqueFieldAPIHelper.java

```diff
@@ -0,0 +1,71 @@
+package com.dotcms.contenttype.business;
+
+import com.dotcms.business.WrapInTransaction;
+import com.dotcms.content.elasticsearch.business.ESContentletAPIImpl;
+import com.dotcms.util.CollectionsUtils;
+import com.dotmarketing.business.FactoryLocator;
+import com.dotmarketing.exception.DotDataException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This Helper allow you to interact with the unique_fields table
+ */
+public class UniqueFieldAPIHelper {
+
+    private final UniqueFieldFactory uniqueFieldFactory;
+
+    public UniqueFieldAPIHelper(){
+        this(FactoryLocator.getUniqueFieldFactory());
+    }
+
+    public UniqueFieldAPIHelper(final UniqueFieldFactory uniqueFieldFactory) {
+        this.uniqueFieldFactory = uniqueFieldFactory;
+    }
+
+
+    /**
+     * Insert a new unique field value, if the value is duplicated then a {@link java.sql.SQLException} is thrown.
+     *
+     * @param uniqueFieldCriteria
+     * @param contentletId
+     *
+     * @throws UniqueFieldValueDupliacatedException when the Value is duplicated
+     * @throws DotDataException when a DotDataException is throws
+     */
+    public void insert(final UniqueFieldCriteria uniqueFieldCriteria, final String contentletId)
+            throws UniqueFieldValueDupliacatedException, DotDataException {
+
```

Comment:
may be add some logs for the support guys in the future

Reply:
https://github.com/dotCMS/core/pull/30332/commits/bc8ee9ff5ad2bf50a2420677772be1feb108f760#diff-949f80e4b1c8d8fcb0be5b680a9e6ec28d60dbaf50ea9e986a6addcdffb7a7b1L41-L61